Helper/Hashcash/HashcashHelper.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+
+namespace EMS\ClientHelperBundle\Helper\Hashcash;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\Security\Csrf\CsrfTokenManager;
+
+class HashcashHelper
+{
+
+    /**
+     * @var CsrfTokenManager
+     */
+    private $csrfTokenManager;
+
+
+
+    public function __construct(CsrfTokenManager $csrfTokenManager)
+    {
+        $this->csrfTokenManager = $csrfTokenManager;
+    }
+
+
+    public function validateHashcash(Request $request, string $csrfId, int $hashcashLevel = 4, string $hashAlgo = 'sha256')
+    {
+        $hashcash = $request->headers->get('X-Hashcash');
+        if ($hashcash === null) {
+            throw new AccessDeniedHttpException('Unrecognized user');
```

Comment:
message should be "hashcash missing", there is not necesarrily a link with hashcash and users.

Reply:
I dont want to say that the hashcash is missing as I do not want that hacker knows that they should investigate that aspect

Reply:
Ok, maybe no Message should be passed?